index.js

```diff
@@ -60,54 +65,60 @@ export default {
         computed: {},
         methods: {},
       };
-
-      // build new objects by removing keys if already exists (e.g. created by mixins)
-      Object.keys(parentData).forEach((e) => {
-        if (typeof $data[e] === 'undefined') {
-          passthrough.$data[e] = parentData[e];
-        }
-      });
-      Object.keys(parentProps).forEach((e) => {
-        if (typeof $props[e] === 'undefined') {
-          passthrough.$props[e] = parentProps[e];
-        }
-      });
-      Object.keys(parentMethods).forEach((e) => {
-        if (typeof methods[e] === 'undefined') {
-          passthrough.methods[e] = parentMethods[e];
-        }
-      });
-      Object.keys(parentComputed).forEach((e) => {
-        if (typeof computed[e] === 'undefined') {
-          passthrough.computed[e] = parentComputed[e];
-        }
-      });
-      Object.keys(parentComponents).forEach((e) => {
-        if (typeof components[e] === 'undefined') {
-          passthrough.components[e] = parentComponents[e];
-        }
-      });
+      
+      if (!this.scoped) {
+        // build new objects by removing keys if already exists (e.g. created by mixins)
+        Object.keys(parentData).forEach((e) => {
+          if (typeof $data[e] === 'undefined') {
+            passthrough.$data[e] = parentData[e];
+          }
+        });
+        Object.keys(parentProps).forEach((e) => {
+          if (typeof $props[e] === 'undefined') {
+            passthrough.$props[e] = parentProps[e];
+          }
+        });
+        Object.keys(parentMethods).forEach((e) => {
+          if (typeof methods[e] === 'undefined') {
+            passthrough.methods[e] = parentMethods[e];
+          }
+        });
+        Object.keys(parentComputed).forEach((e) => {
+          if (typeof computed[e] === 'undefined') {
+            passthrough.computed[e] = parentComputed[e];
+          }
+        });
+        Object.keys(parentComponents).forEach((e) => {
+          if (typeof components[e] === 'undefined') {
+            passthrough.components[e] = parentComponents[e];
+          }
+        });
+      }
 
       const methodKeys = Object.keys(passthrough.methods || {});
       const dataKeys = Object.keys(passthrough.$data || {});
       const propKeys = Object.keys(passthrough.$props || {});
       const templatePropKeys = Object.keys(this.templateProps);
-      const allKeys = dataKeys.concat(propKeys).concat(methodKeys).concat(templatePropKeys);
+      const allKeys = this.scoped ? templatePropKeys : dataKeys.concat(propKeys).concat(methodKeys).concat(templatePropKeys);
       const methodsFromProps = buildFromProps(parent, methodKeys);
       const finalProps = merge([
         passthrough.$data,
         passthrough.$props,
         methodsFromProps,
         this.templateProps,
       ]);
+      const components = this.scoped ? this.templateComponents : {
```

Comment:
`components` is already defined earlier, so my compiler to bugging out on this. Can you tweak this?